packages/client/src/components/notebook/AddVariablePopover.tsx

```diff
@@ -475,8 +479,19 @@ export const AddVariablePopover = observer((props: AddVariablePopoverProps) => {
                             </Alert>
                         );
                     }
-                } else if (inputVariableTypeList.indexOf(variableType) > -1) {
-                    return JSON.stringify(variableInputValue);
+                } else if (
+                    variableType === 'JSON' ||
```

Comment:
If we edit a variable, seems like it isn't respecting this if statement

Reply:
Might be a good place to put your utility function

Reply:
Confirmed that the statement works and added in utility function.

Reply:
think we can also clean up these conditions

Reply:
Added back original condition, now works with all variable types including json / array.